boot/freeldr/freeldr/arch/i386/pc/pcdisk.c

```diff
@@ -586,14 +586,19 @@ PcDiskReadLogicalSectorsLBA(
      * Disk address packet's block count field set to the
      * number of blocks successfully transferred.
      */
-    RegsIn.b.ah = 0x42;                 // Subfunction 42h
+    RegsIn.b.ah = 0x42;
     RegsIn.b.dl = DriveNumber;          // Drive number in DL (0 - floppy, 0x80 - harddisk)
     RegsIn.x.ds = BIOSCALLBUFSEGMENT;   // DS:SI -> disk address packet
     RegsIn.w.si = BIOSCALLBUFOFFSET;
 
     /* Retry 3 times */
     for (RetryCount = 0; RetryCount < 3; ++RetryCount)
     {
+        /* Restore the number of blocks to transfer, since it gets reset
+         * on failure with the number of blocks that were successfully
+         * transferred (and which could be zero). */
+        Packet->LBABlockCount = (USHORT)SectorCount;
```

Comment:
Maybe comment out / delete / move lines 573-574? Or move new code to line 613?

Note:
`else` and `continue` are superfluous in following existing code.

Reply:
> Note: `else` and `continue` are superfluous in following existing code.

Submitted as  #7417.